packages/core/src/delegation/DelegationNode.ts

```diff
@@ -273,25 +270,25 @@ export class DelegationNode implements IDelegationNode {
     delegateDid: DidDetails,
     sign: SignCallback,
     {
-      keySelection = DidUtils.defaultKeySelectionCallback,
+      keySelection = Did.Utils.defaultKeySelectionCallback,
     }: {
       keySelection?: DidKeySelectionCallback<DidVerificationKey>
     } = {}
-  ): Promise<DidUtils.EncodedSignature> {
-    const authenticationKey = await keySelection(
-      delegateDid.getVerificationKeys('authentication')
-    )
+  ): Promise<Did.Utils.EncodedSignature> {
+    // TODO: do we need keySelection for the single authentication key here and elsewhere?
```

Comment:
I fear we have to, because on-chain it is stored as an array. Not sure, if we currently allow having multiple authentication keys though. In that case we might be able to remove the keyselection for the time being.

Reply:
> on-chain it is stored as an array

I don’t think this is true, and [William confirmed my understanding](https://botlabs-group.slack.com/archives/CKB1ZF0QY/p1660894109310849)

Reply:
ah, good to know. I also remember some conversations around this now.
I think we should get rid of multiple keys for all the verification keys than. Makes it much simpler to use.
That would work according to w3c spec and would reflect the blockchain, too.